### PR TITLE
{Release} Upgrade to Azure CLI 2.51.0

### PR DESCRIPTION
**Pull request by azclibot**
Updated Azure CLI version.

### CODE REVIEW

**Comment by yonzhan** [comment]
Release

**Review by jiasli — DISMISSED** [review]

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,103 @@
...
 ===============
 
+2.51.0
+++++++
+
+**AKS**
+
+* `az aks nodepool snapshot update`: Add aks nodepool snapshot update command (#26790)
+* `az aks create`: Add new parameter `--k8s-support-plan` to support LTS onboarding, also add new tier enum `premium` (#26795)
+* `az aks update`: Support to enable/disable LTS via new parameter `--k8s-support-plan` (#26795)
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:13]:
```suggestion
* `az aks update`: Support enabling/disabling LTS via new parameter `--k8s-support-plan` (#26795)
```

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,103 @@
...
+* Fix #21133: `az webapp/functionapp config ssl bind/unbind`: Search for matching certificates in the subscription by App Service Plan Id (#26617)
+
+**ARM**
+
+* `az stack`: Fix the bug that the required `--deny-settings-mode` parameter should not return None (should be a string) (#26900)
+* `az stack`: Fix the bug that the `--deny-settings-excluded-principals` parameter was accidentally being reset (#26900)
+
+**Batch**
+
+* Remove `az batch job all-statistics` and `az batch pool all-statistics` which no longer worked in the past (#26766)
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:45]:
```suggestion
* `az batch job/pool all-statistics`: Remove no longer worked commands (#26766)
```

File: src/azure-cli/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,103 @@
...
+* `az batch pool create`: Add new parameter `--enable-accelerated-networking` to determine whether this pool should enable accelerated networking (#26766)
+
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:47]:
```suggestion

**Cognitive Services**

* `az cognitiveservices account deployment create`: Add `--sku-name` and `--sku-capacity` parameters (#26995)
* `az cognitiveservices usage`: Add new command `list` (#26995)
* `az cognitiveservices model`: Add new command `list` (#26995)

```

File: src/azure-cli/HISTORY.rst [hunk 4]
```diff
@@ -3,6 +3,109 @@
...
+* `az aks nodepool snapshot update`: Add aks nodepool snapshot update command (#26790)
+* `az aks create`: Add new parameter `--k8s-support-plan` to support LTS onboarding, also add new tier enum `premium` (#26795)
+* `az aks update`: Support to enable/disable LTS via new parameter `--k8s-support-plan` (#26795)
+* `az aks create`: Add node taint support when create cluster use `az aks` command (#26837)
+* `az aks update`: Add update node taint support on cluster level use `az aks` command (#26837)
+* `az aks enable-addons`: Fix the default value of option `--enable-msi-auth-for-monitoring` being overwritten to `False` when specified (#26844)
+* `az aks update`: Add new parameter `--outbound-type` to support cluster outbound type. (#26975)
+* `az aks maintenanceconfiguration list`: Add new command to list all maintenance windows in a cluster (#26758)
+* `az aks maintenanceconfiguration show`: Add new command to display a specific maintenance windows of a cluster (#26758)
```
Comment (yanzhudd) [inline src/azure-cli/HISTORY.rst:19]:
```suggestion
* `az aks maintenanceconfiguration show`: Add new command to display a specific maintenance window of a cluster (#26758)
```

File: src/azure-cli/HISTORY.rst [hunk 5]
```diff
@@ -3,6 +3,109 @@
...
+* `az cognitiveservices account deployment create`: Add `--sku-name` and `--sku-capacity` parameters (#26995)
+* `az cognitiveservices usage`: Add new command `list` (#26995)
+* `az cognitiveservices model`: Add new command `list` (#26995)
+
+**Compute**
+
+* `az vm/vmss create`: Enable auto upgrade of guest attestation extension by default for Trusted Launch enabled VMs and VMSS (#26878)
+* `az vm/vmss create`: Add new parameter `--disable-integrity-monitoring-autoupgrade` to support disabling auto upgrade of guest attestation extension for Trusted Launch enabled VMs and VMSS (#26878)
+* `az sig image-version undelete`: Add new command to support softdeleted image recovery (#26943)
+* Add Standard as SecurityType Option for backward compatibility (#26892)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:59]:
```suggestion
* `az vm/vmss/disk create`: Add new option `Standard` for `--security-type` for backward compatibility (#26892)
* `az sig image-definition create`: Add new option `Standard` for `--security-type` for backward compatibility (#26892)
```

File: src/azure-cli/HISTORY.rst [hunk 6]
```diff
@@ -3,6 +3,110 @@
...
+
+**Cognitive Services**
+
+* `az cognitiveservices account deployment create`: Add `--sku-name` and `--sku-capacity` parameters (#26995)
+* `az cognitiveservices usage`: Add new command `list` (#26995)
+* `az cognitiveservices model`: Add new command `list` (#26995)
+
+**Compute**
+
+* `az vm/vmss create`: Enable auto upgrade of guest attestation extension by default for Trusted Launch enabled VMs and VMSS (#26878)
```
Comment (yanzhudd) [inline src/azure-cli/HISTORY.rst:56]:
```suggestion
* `az vm/vmss create`: Enable auto upgrading of guest attestation extension by default for Trusted Launch enabled VMs and VMSS (#26878)
```

File: src/azure-cli/HISTORY.rst [hunk 7]
```diff
@@ -3,6 +3,110 @@
...
+* `az vm/vmss create`: Add new parameter `--disable-integrity-monitoring-autoupgrade` to support disabling auto upgrade of guest attestation extension for Trusted Launch enabled VMs and VMSS (#26878)
```
Comment (yanzhudd) [inline src/azure-cli/HISTORY.rst:57]:
```suggestion
* `az vm/vmss create`: Add new parameter `--disable-integrity-monitoring-autoupgrade` to support disabling auto upgrading of guest attestation extension for Trusted Launch enabled VMs and VMSS (#26878)
```

File: src/azure-cli/HISTORY.rst [hunk 8]
```diff
@@ -3,6 +3,109 @@
...
+**App Config**
+
+* `az appconfig kv delete/set/set-keyvault`: Add key validations for null or empty space keys (#26928)
+* `az appconfig kv export/import/restore`: Update key-value diffing and preview (#26325)
+* `az appconfig snapshot`: Remove status code property from snapshot object (#26891)
+* `az appconfig snapshot list`: Use enums for status parameter (#26879)
+
+**App Service**
+
+* Fix #26214: Missing leading slash causes web app and plan commands to fail for s-clouds (#26921)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:34]:
```suggestion
* Fix #26214: `az webapp config container set`: Fix the bug caused by Missing leading slash causes web app and plan commands to fail for s-clouds (#26921)
```

File: src/azure-cli/HISTORY.rst [hunk 9]
```diff
@@ -3,6 +3,109 @@
...
+* Fix #26601: `az functionapp create`: Throw error for consumption function app created with vnet (#26792)
+* Fix #21133: `az webapp/functionapp config ssl bind/unbind`: Search for matching certificates in the subscription by App Service Plan Id (#26617)
+
+**ARM**
+
+* `az stack`: Fix the bug that the required `--deny-settings-mode` parameter should not return None (should be a string) (#26900)
+* `az stack`: Fix the bug that the `--deny-settings-excluded-principals` parameter was accidentally being reset (#26900)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:41]:
```suggestion
* `az stack`: Fix the bug that the `--deny-settings-excluded-principals` parameter was accidentally reset (#26900)
```

File: src/azure-cli/HISTORY.rst [hunk 10]
```diff
@@ -3,6 +3,110 @@
...
+
+* `az vm/vmss create`: Enable auto upgrading of guest attestation extension by default for Trusted Launch enabled VMs and VMSS (#26878)
+* `az vm/vmss create`: Add new parameter `--disable-integrity-monitoring-autoupgrade` to support disabling auto upgrading of guest attestation extension for Trusted Launch enabled VMs and VMSS (#26878)
+* `az sig image-version undelete`: Add new command to support softdeleted image recovery (#26943)
+* `az vm/vmss/disk create`: Add new option `Standard` for `--security-type` for backward compatibility (#26892)
+* `az sig image-definition create`: Add new option `Standard` for `--security-type` for backward compatibility (#26892)
+
+**Cosmos DB**
+
+* `az cosmosdb restore`: Add `--assign-identity` and `--default-identity` to allow PITR restore with identity (#26867)
```
Comment (yanzhudd) [inline src/azure-cli/HISTORY.rst:64]:
```suggestion
* `az cosmosdb restore`: Add `--assign-identity` and `--default-identity` to allow PITR restoring with identity (#26867)
```

File: src/azure-cli/HISTORY.rst [hunk 11]
```diff
@@ -3,6 +3,110 @@
...
+* `az cosmosdb postgres`: Add new command groups to support Cosmos DB for PostgreSQL (#26729)
+
+**Key Vault**
+
+* `az keyvault restore start`: Add `--key-name` to support selective key restore (#26907)
```
Comment (yanzhudd) [inline src/azure-cli/HISTORY.rst:69]:
```suggestion
* `az keyvault restore start`: Add `--key-name` to support selective key restoring (#26907)
```

File: src/azure-cli/HISTORY.rst [hunk 12]
```diff
@@ -3,6 +3,110 @@
...
+**App Config**
+
+* `az appconfig kv delete/set/set-keyvault`: Add key validations for null or empty space keys (#26928)
+* `az appconfig kv export/import/restore`: Update key-value diffing and preview (#26325)
+* `az appconfig snapshot`: Remove status code property from snapshot object (#26891)
+* `az appconfig snapshot list`: Use enums for status parameter (#26879)
+
+**App Service**
+
+* Fix #26214: `az webapp config container set`: Fix the bug caused by missing leading slash causes web app and plan commands to fail for s-clouds (#26921)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:34]:
```suggestion
* Fix #26214: `az webapp show`: Fix the bug caused by missing leading slash causes web app and plan commands to fail for s-clouds (#26921)
* Fix #26214: `az appservice plan show`: Fix the bug caused by missing leading slash causes web app and plan commands to fail for s-clouds (#26921)
```

File: src/azure-cli/HISTORY.rst [hunk 13]
```diff
@@ -3,6 +3,110 @@
...
+* `az functionapp connection`: Add new command group to support service connector on Function App (#26825)
+* `az spring connection`: Enable new auth types for Spring Boot and Cosmos SQL connection (#26719)
+
+**SQL**
+
+* `az sql mi start/stop/start-stop-schedule`: Add SQL MI manual and scheduled start stop (#26979)
+
+**Storage**
+
+* `az storage container-rm update`: `--default-encryption-scope` and `--deny-encryption-scope-override` should not be specified during update (#26948)
```
Comment (yanzhudd) [inline src/azure-cli/HISTORY.rst:107]:
```suggestion
* `az storage container-rm update`: `--default-encryption-scope` and `--deny-encryption-scope-override` should not be specified during updating (#26948)
```

**Review by evelyn-ys — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Review by wangzelin007 — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Pull request merged**